Dockerfile

```diff
@@ -1,22 +1,34 @@
-FROM node:14-alpine AS BUILD_IMAGE
-
-COPY package.json yarn.lock /
-
+# Install dependencies only when needed
+FROM node:alpine AS deps
+# Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
+RUN apk add --no-cache libc6-compat
+WORKDIR /app
+COPY package.json yarn.lock ./
 RUN yarn install --frozen-lockfile
 
+# Rebuild the source code only when needed
+FROM node:alpine AS builder
+WORKDIR /app
 COPY . .
+COPY --from=deps /app/node_modules ./node_modules
+RUN yarn build && yarn install --production --ignore-scripts --prefer-offline
 
-RUN yarn build
+# Production image, copy all the files and run next
+FROM node:alpine AS runner
+WORKDIR /app
 
-FROM nginx:alpine
+ENV NODE_ENV production
 
-WORKDIR /usr/share/nginx/html
+RUN addgroup -g 1001 -S nodejs
+RUN adduser -S nextjs -u 1001
 
-COPY --from=BUILD_IMAGE /build .
+# You only need to copy next.config.js if you are NOT using the default configuration
+# COPY --from=builder /app/next.config.js ./
+COPY --from=builder /app/public ./public
+COPY --from=builder --chown=nextjs:nodejs /app/.next ./.next
+COPY --from=builder /app/node_modules ./node_modules
+COPY --from=builder /app/package.json ./package.json
 
-COPY nginx.conf /etc/nginx/conf.d/default.conf
+USER nextjs
 
-ARG BUILDTIME
-ARG COMMITHASH
-ENV BUILDTIME ${BUILDTIME}
-ENV COMMITHASH ${COMMITHASH}
```

Comment:
@bodymindarts this is part of our CI deployment?

Reply:
Yes we want to preserve this! Can simply add these layer onto the default next image.

Reply:
I added those back, but will this work?

Reply:
I added a `CMD` to the dockerfile and tested it locally, and I do get the page working. Cannot test lnurl as the hostname locally is localhost.